src/dmd/doc.d

```diff
@@ -1920,6 +1920,74 @@ unittest {
     assert(skipChars(&buf, 12, "dne") == 15);
 }
 
+/****************************************************
+ * Escape the given string
+ * Params:
+ *  s = the string to escape
+ *  c = the character to look for
+ *  r = the string to replace `c` with
+ * Returns: `s` with `c` replaced with `r`
+ */
+private string escapeString(string s, char c, string r) pure
+{
+    for (size_t i = 0; i < s.length; ++i)
+    {
+        if (s[i] == c)
+        {
+            s = s[0..i] ~ r ~ s[i+1..$];
+            i += r.length-1;
+        }
+    }
+    return s;
+}
+
+///
+unittest
+{
+    assert("".escapeString(',', "$(COMMA)") == "");
+    assert("ab".escapeString(',', "$(COMMA)") == "ab");
+    assert("a,b".escapeString(',', "$(COMMA)") == "a$(COMMA)b");
+    assert("a,,b".escapeString(',', "$(COMMA)") == "a$(COMMA)$(COMMA)b");
+    assert(",ab".escapeString(',', "$(COMMA)") == "$(COMMA)ab");
+    assert("ab,".escapeString(',', "$(COMMA)") == "ab$(COMMA)");
+}
+
+/**
+ * Return a lowercased copy of a string.
+ * Params:
+ *  s = the string to lowercase
+ * Returns: the lowercase version of the string or the original if already lowercase
+ */
+private string toLowercase(string s) pure
+{
+    string lower;
```

Comment:
You already know the length of the string here, so you could allocate once and skip all the concatenations below.

Reply:
I made that change.